spec/fabricators/access_token_fabricator.rb

```diff
@@ -1,3 +1,7 @@
 # frozen_string_literal: true
 
-Fabricator :access_token, from: 'Doorkeeper::AccessToken'
+Fabricator :access_token, from: 'Doorkeeper::AccessToken' do
+  # This is necessary as for some reason Fabricator doesn't pick up the setting
+  # from Doorkeeper automatically:
+  use_refresh_token { Doorkeeper.config.refresh_token_enabled? }
```

Comment:
I'm really surprised at this, this might require digging a little deeper.

Reply:
Yeah, I was surprised by it too! I'm guessing it's just some sort of weird quirk with how Doorkeeper works:
- The `use_refresh_token` isn't inherent to the model mixin: https://github.com/doorkeeper-gem/doorkeeper/blob/main/lib/doorkeeper/models/access_token_mixin.rb#L182
- It seems to be passed into the model, rather than the config being fetched by the model: https://github.com/doorkeeper-gem/doorkeeper/blob/main/lib/doorkeeper/oauth/base_request.rb#L38

Unlike scopes or token reuse that seems to be in the model itself.

There is also a potential risk here in that Doorkeeper's Refresh Tokens don't actually expire, when they absolutely should: https://github.com/doorkeeper-gem/doorkeeper/wiki/Enable-Refresh-Token-Credentials

Essentially Doorkeeper models refresh tokens as just a string on the Access Token, rather than having Access Tokens having one Refresh Token. Maybe we need to see if we can make upstream changes to Doorkeeper to have refresh tokens be actual tokens / records rather than just strings on Access Tokens.

Ideally what you have is a "token family", where the initial Access Token is issued from a grant flow, and it references that flow, and then the refresh token essentially daisy chains off that, so you've: `Authorization Grant -> Access Token (1) -> Refresh Token -> Access Token (2)`. This allows for security measures where if you detect the Refresh Token on Access Token (1) is reused, then you revoke the entire family of tokens, since that family can be deemed to be compromised, this is what's recommended by Auth0: https://auth0.com/blog/refresh-tokens-what-are-they-and-when-to-use-them/#Refresh-Token-Automatic-Reuse-Detection

Reply:
Apparently Doorkeeper's Access Grants also aren't related in any way to the Access Token's that they generate.. which is rather strange!